src/MotionHandler.moon

```diff
@@ -167,25 +167,79 @@ class MotionHandler
 
 				local newText
 				if @options.main.killTrans
-					newText = line\interpolateTransformsCopy timeDelta, newStartTime
+					newText = \interpolateTransformsCopy timeDelta, newStartTime
 				else
-					newText = line\detokenizeTransformsCopy timeDelta
-
-				newText = newText\gsub "\\fade(%b())", ( fade ) ->
-					a1, a2, a3, t1, t2, t3, t4 = fade\match("(%d+),(%d+),(%d+),(%d+),(%d+),(%d+),(%d+)")
-					if t1 == nil
-						message = "There is a malformed \\fade you must fix.\n\\fade requires 7 integer arguments.\nLine: #{.number}, tag: \\fade(#{fade})."
-						if fade\match("(%d+),(%d+)")
-							message ..= "\nPerhaps you meant to use \\fad."
-						log.windowError message
-
-					t1, t2, t3, t4 = tonumber( t1 ), tonumber( t2 ), tonumber( t3 ), tonumber( t4 )
-					-- beautiful.
-					t1 -= timeDelta
-					t2 -= timeDelta
-					t3 -= timeDelta
-					t4 -= timeDelta
-					("\\fade(%s,%s,%s,%d,%d,%d,%d)")\format a1, a2, a3, t1, t2, t3, t4
+					newText = \detokenizeTransformsCopy timeDelta
+
+				fadeTag = tags.allTags.fade
+				if @options.main.killTrans
+					local fade
+					newText = newText\gsub "({.-})", ( tagBlock ) -> tagBlock\gsub fadeTag.pattern, ( value ) ->
+						fade = fadeTag\convert value
+						return ""
+
+					if fade
+						-- makes sure every necessary alpha tag exists in the first block
+						alpha = tags.allTags.alpha
+						alpha1, alpha2, alpha3, alpha4 = tags.allTags.alpha1, tags.allTags.alpha2, tags.allTags.alpha3, tags.allTags.alpha4
+						outline, shadow = tags.allTags.border, tags.allTags.shadow
+						newText = newText\gsub "^({.-})", ( tagBlock ) ->
```

Comment:
This is not the right place to do this. The alpha tags should be substituted in somewhere in `prepareLines` in `aegisub-motion.moon`. In here, it's running this text substitution for every frame of every line. In there it would only run once for every line.